### PR TITLE
fix: Clean up deprecated functions 

### PR DESCRIPTION
**Pull request by marioferh**
- Clean up deprecated funcs
- Minor clean ups

### CODE REVIEW

File: test/e2e/monitoring_stack_controller_test.go [hunk 1]
```diff
@@ -49,7 +49,10 @@ func assertCRDExists(t *testing.T, crds ...string) {
 }
 
 func TestMonitoringStackController(t *testing.T) {
-	stack.AddToScheme(scheme.Scheme)
+	err := stack.AddToScheme(scheme.Scheme)
+	if err != nil {
+		return
+	}
```
Comment (sthaha) [inline test/e2e/monitoring_stack_controller_test.go:55]:
```suggestion
	err := stack.AddToScheme(scheme.Scheme)
	assert.NilError(t, err, "adding stack to scheme failed")
```

Comment (marioferh) [reply test/e2e/monitoring_stack_controller_test.go]: thanks!

**Review by sthaha — COMMENTED** [review]
/lgtm
I think there is no harm in waiting for a minute.

File: test/e2e/framework/assertions.go [hunk 2]
```diff
@@ -105,17 +104,17 @@ func (f *Framework) AssertResourceEventuallyExists(name, namespace string, resou
...
+		WaitTimeout:  DefaultTestTimeout,
 	}
 	for _, fn := range fns {
 		fn(&option)
 	}
 	return func(t *testing.T) {
 		key := types.NamespacedName{Name: name, Namespace: namespace}
 		//nolint
-		if err := wait.Poll(5*time.Second, option.WaitTimeout, func() (bool, error) {
+		if err := wait.Poll(5+time.Second, option.WaitTimeout, func() (done bool, err error) {
```
Comment (simonpasquier) [inline test/e2e/framework/assertions.go:115]: why this change while we're not replacing wait.Poll?

File: test/e2e/monitoring_stack_controller_test.go [hunk 3]
```diff
@@ -49,7 +49,8 @@ func assertCRDExists(t *testing.T, crds ...string) {
 }
 
 func TestMonitoringStackController(t *testing.T) {
-	stack.AddToScheme(scheme.Scheme)
+	err := stack.AddToScheme(scheme.Scheme)
+	assert.NilError(t, err, "adding stack to scheme failed")
```
Comment (simonpasquier) [inline test/e2e/monitoring_stack_controller_test.go:53]: :+1:

File: test/e2e/thanos_querier_controller_test.go [hunk 4]
```diff
@@ -139,31 +137,28 @@ func newThanosQuerier(t *testing.T, name string, selector map[string]string) *ms
 }
 
 func waitForThanosQuerierDeletion(tq *msov1.ThanosQuerier) error {
-	//nolint
-	return wait.Poll(5*time.Second, wait.ForeverTestTimeout, func() (bool, error) {
-		err := f.K8sClient.Get(context.Background(),
+	return wait.PollUntilContextTimeout(context.Background(), 5*time.Second, wait.ForeverTestTimeout, true, func(ctx context.Context) (done bool, err error) {
```
Comment (simonpasquier) [inline test/e2e/thanos_querier_controller_test.go:140]: General question: I see that we replace some function signatures with named parameters but not all of them, is there a concrete motivation? 
Personally I'm not a big fan of named parameters but I think that we need to be consistent.

File: test/e2e/framework/assertions.go (hunk 2 above)
Comment (marioferh) [reply test/e2e/framework/assertions.go]: typo

File: test/e2e/monitoring_stack_controller_test.go (hunk 3 above)
Comment (marioferh) [reply test/e2e/monitoring_stack_controller_test.go:53]: :D

File: test/e2e/thanos_querier_controller_test.go (hunk 4 above)
Comment (marioferh) [reply test/e2e/thanos_querier_controller_test.go:140]: wait.Poll is deprecated in this version, tried to change for PollUntilContextTimeout all over the code but in a couple of funcs there are issues with context, then I let wait.Poll with //nolint and I'll try to fix in next iteration.

Comment (marioferh) [reply test/e2e/thanos_querier_controller_test.go:140]: in library examples PollUntilContextTimeout is done in this way, I'll change to named parameters in all funcs. thanks

**Review by simonpasquier — APPROVED** [review]
thanks!

File: test/e2e/framework/assertions.go [hunk 5]
```diff
@@ -23,7 +23,9 @@ import (
 	"k8s.io/apimachinery/pkg/util/wait"
 )
 
-const CustomForeverTestTimeout = 40 * time.Second
+// default ForeverTestTimeout is 30, some test fails because they take more than 30s
```
Comment (danielmellado) [inline test/e2e/framework/assertions.go:26]: Minor typo ;)

```suggestion
// default ForeverTestTimeout is 30, some test fail because they take more than 30s
```

File: test/e2e/framework/assertions.go [hunk 6]
```diff
@@ -44,19 +46,19 @@ func WithPollInterval(d time.Duration) OptionFn {
...
+		WaitTimeout:  DefaultTestTimeout,
 	}
 	for _, fn := range fns {
 		fn(&option)
 	}
 
 	return func(t *testing.T) {
 		//nolint
-		if err := wait.Poll(option.PollInterval, option.WaitTimeout, func() (done bool, err error) {
+		if err := wait.Poll(option.PollInterval, option.WaitTimeout, func() (bool, error) {
```
Comment (danielmellado) [inline test/e2e/framework/assertions.go:61]: Is there any reason on why the named parameters have been dropped? IMHO they help on code readability.

**Review by danielmellado — COMMENTED** [review]
/lgtm, just some minor nits

File: test/e2e/framework/assertions.go (hunk 6 above)
Comment (marioferh) [reply test/e2e/framework/assertions.go:61]: before my change every wait.Poll was with no named parameters. Just to maintain the coherence.
